### PR TITLE
fixes: size of group members for 'non-admin group viewer' should be less

### LINKED ISSUE #2582: fix: Size of group members for 'non-admin group viewer' should be less.
**Describe the bug**
fix: Size of group members for 'non-admin group viewer' should be less

**To Reproduce**
1. Go to 'My Groups'
2. Click on 'view'
3. Scroll down to 'members'

**Expected behavior**
- For now the embed ruby if-else condition generate div for "remove" button that should not be created.
- For non-admin user in the class 'group-members-card' should have width:200px

**Screenshots**
bandicam 2021-11-03 21-05-30-648

**Device Information [optional]:**

**Additional context**

**Are you working on this issue? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by vedant-jain03**
Fixes #2582 

#### Describe the changes you have made in this PR -
- Added a new class for 'groups-members-card-viewer' where the width: 200px and height: auto
- If the visiter is not admin the grid is now divided into 3 but it should be 4 according to width size.
- embed ruby if condition before remove-button div.

### Screenshots of the changes (If any) -
- Before
bandicam 2021-11-03 21-15-24-473

- After
bandicam 2021-11-03 21-15-39-876

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by vedant-jain03** [comment]
@gr455 @tachyons @nitin10s Kindly Review.

**Review by nitin10s — CHANGES_REQUESTED** [review]
Looks good, can we increase the margin between the icon and the name. Also, please slightly increase the left padding from the icon. 

The icon doesn't appear vertically centered, can we fix that?

**Comment by vedant-jain03** [comment]
> Looks good, can we increase the margin between the icon and the name. Also, the icon doesn't appear vertically centered, can we fix that?

Yess Sure.

**Comment by vedant-jain03** [comment]
Screenshot (171)

1. Added margin-right: 20px
2. Make user icon center
3. Added tooltiptext feature for showing the full name of the user in hover.

@nitin10s Kindly Review.

**Comment by vedant-jain03** [comment]
Screenshot (174)

1. I have added a new class for non-admin-user `groups-members-card-details-non-user` as we don't want to messup with `groups-members-card-details` class and added a conditional check to implement a class for the respective user.
2. Added margin-right for user icon 15px.
3. Added tooltiptext for better visibility of username as their is `text-overflow: ellipse`.
4. center user-icon.

@nitin10s Kindly Review.

**Review by nitin10s — CHANGES_REQUESTED** [review]
There is a lot of space in the right that we can utilize before using overflow: ellipsis.

**Comment by nitin10s** [comment]
Everything else looks great!

**Comment by vedant-jain03** [comment]
> There is a lot of space in the right that we can utilize before using overflow: ellipsis.

Sure

**Comment by vedant-jain03** [comment]
Screenshot (177)

Change `col-7` to `col-11` for better space utilization.

**Comment by vedant-jain03** [comment]
> There is a lot of space in the right that we can utilize before using overflow: ellipsis.

Kindly Review @nitin10s

**Comment by nitin10s** [comment]
The last thing, why do the name blocks don't start from the extreme left aligning to the members' headline?

**Comment by vedant-jain03** [comment]
> The last thing, why do the name blocks don't start from the extreme left aligning to the members' headline? Can we please fix that?

@nitin10s I think that is because of gap between user icon and name, as you proposed in this : https://github.com/CircuitVerse/CircuitVerse/pull/2583#pullrequestreview-805636606
If you want I can decrease the gap as well ?

**Comment by vedant-jain03** [comment]
> The last thing, why do the name blocks don't start from the extreme left aligning to the members' headline? Can we please fix that?

Sure we can fix that 👍🏼

**Comment by vedant-jain03** [comment]
bandicam 2021-11-30 21-08-03-298

@nitin10s Kindly review

**Review by nitin10s — APPROVED** [review]
@tachyons we can merge

File: app/views/groups/_group_members.html.erb [hunk 1]
```diff
@@ -1,24 +1,25 @@
 <% group.users.each do |user| %>
-  <div class="col-xs-12 col-sm-12 col-md-6 col-lg-4 groups-members-card-container">
-    <div class="groups-members-card">
+  <div class="<%= policy(group).admin_access? ? 'col-xs-12 col-sm-12 col-md-6 col-lg-4 groups-members-card-container' : 'col-xs-12 col-sm-12 col-md-6 col-lg-3 groups-members-card-container' %>">
```
Comment (tachyons) [inline app/views/groups/_group_members.html.erb:2]: Consider moving this to a helper as this is being used multiple places

Comment (vedant-jain03) [reply app/views/groups/_group_members.html.erb:2]: Sure

**Comment by vedant-jain03** [comment]
@tachyons Kindly review

File: app/views/groups/_group_members.html.erb (hunk 1 above)
Comment (vedant-jain03) [reply app/views/groups/_group_members.html.erb]: @tachyons Added the helpers

**Review by gr455 — APPROVED** [review]
Looks good to me. @tachyons You can merge this after you take a final look

**Pull request merged**